components/teachers/teacher.js

```diff
@@ -39,128 +44,134 @@ function Teacher({log, setHours, setTeacher, invalidate, teacher, courses, data,
+                                                key={course.name + "-input4"}
+                                                className={warning("period", "p4", course) ? `${styles.warning}` : `${styles.inputBase}`}
+                                                onChange={event => modifyHours(event, course.name, "p4")}
+                                                onKeyDown={numbersOnly}
+                                                defaultValue={course.hours.p4}
+                                                margin={"dense"}
+                                                inputProps={{"aria-label": "naked"}} />
+                                        </TableCell>
+                                    </TableRow>
+                                </Fragment>
+                            )
+                        })}
+                        <TableRow
+                            key={teacherFullName + "-periodHoursRow"}
+                            className={styles.tableRow}>
+                            <TableCell
+                                key={teacherFullName + "-cell1"}
+                                className={warning("totalHours", null, calcTotalHours(periodTotalHours)) ? `${styles.warning}` : (`${styles.tableCell} ${styles.thCustomWidth}`)}
+                                component={"th"}
+                                scope={"row"}>
+                                <strong>{`${strings.totalHours} ${calcTotalHours(periodTotalHours)}`}</strong>
+                            </TableCell>
+                            <TableCell>{periodTotalHours[0]}</TableCell>
+                            <TableCell>{periodTotalHours[1]}</TableCell>
+                            <TableCell>{periodTotalHours[2]}</TableCell>
+                            <TableCell>{periodTotalHours[3]}</TableCell>
+                        </TableRow>
+                    </TableBody>
                 </Table>
             </CardContent>
-            <CardActions className={ctStyles.cardActions}>
-                <AddCourse
-                    setTeacher={setTeacher}
-                    addCourse={invalidate}
-                    teacher={teacherFullName}
-                    dropdownList={courses}
-                    strings={strings}/>
-            </CardActions>
+            {/*            <div className={warning("teacherCourses", null, assignedCourses) ? `${styles.teacherWarning}` : null}> */}
+            <span className={warning("teacherCourses", null, assignedCourses) ? `${styles.warning} ${styles.thCustomWidth}` : null}>
+                <CardActions className={ctStyles.cardActions}>
+                    <AddCourse
+                        setTeacher={setTeacher}
+                        addCourse={invalidate}
+                        teacher={teacherFullName}
+                        dropdownList={courses}
+                        strings={strings} />
+                </CardActions>
+            </span>
```

Comment:
Compromising the semantic structure here by wrapping `CardActions` in a `span` seems unnecessary.